L1Trigger/TrackFindingTMTT/src/Sector.cc

```diff
@@ -0,0 +1,350 @@
+      float tolerancePhiTrk =
+          assumedPhiTrkRes_ * (2 * sectorHalfWidth_);  // Set tolerance equal to nominal resolution assumed in phiTrk
+      if (calcPhiTrkRes_) {
+        // Calculate uncertainty in phiTrk due to poor resolution in stub bend
+        float phiTrkRes = stub->trkPhiAtRres(chosenRofPhi_);
+        // Reduce tolerance if this is smaller than the nominal assumed resolution.
+        tolerancePhiTrk = min(tolerancePhiTrk, phiTrkRes);
+      }
+      float outsidePhiTrk = fabs(delPhiTrk) - sectorHalfWidth_ -
+                            tolerancePhiTrk;  // If > 0, then stub is not compatible with being inside this sector.
+
+      // Modify algorithm to allow for uncertainty due to 2S module strip length, if requested.
+      if (handleStripsPhiSec_) {
+        float chosenStubPhiErr =
+            stub->trkPhiAtR(chosenRofPhi_).second;  // The "Err" here is uncertainty due to 2S strip length.
+        outsidePhiTrk -= chosenStubPhiErr;
+      }
+
+      if (outsidePhiTrk > 0)
+        okPhiTrk = false;
+    }
+
+    return (okPhi && okPhiTrk);
+  }
+
+  //=== For performance studies, note which stubs on given tracking particle are inside the sector.
+  //=== Returns two booleans for each stub, indicating if they are in phi & eta sectors respectively.
+  //=== AND them together to get (eta,phi) sector decision.
+
+  unordered_map<const Stub*, pair<bool, bool> > Sector::stubsInside(const TP& tp) const {
+    unordered_map<const Stub*, pair<bool, bool> > inside;
+    // Loop over stubs produced by tracking particle
+    const vector<const Stub*>& assStubs = tp.assocStubs();
+    for (const Stub* stub : assStubs) {
+      // Check if this stub is inside sector
+      inside[stub] = pair<bool, bool>(this->insidePhi(stub), this->insideEta(stub));
+    }
+    return inside;
+  }
+
+  //=== Count number of stubs in given tracking particle which are inside this (phi,eta) sector;
+  //=== or inside it if only the eta cuts are applied; or inside it if only the phi cuts are applied.
+  //=== The results are returned as the 3 last arguments of the function.
+
+  void Sector::numStubsInside(const TP& tp,
+                              unsigned int& nStubsInsideEtaPhi,
+                              unsigned int& nStubsInsideEta,
+                              unsigned int& nStubsInsidePhi) const {
+    nStubsInsideEtaPhi = 0;
+    nStubsInsideEta = 0;
+    nStubsInsidePhi = 0;
+    for (const auto& iter : this->stubsInside(tp)) {
+      bool insidePhi = iter.second.first;
+      bool insideEta = iter.second.second;
+      if (insidePhi && insideEta)
+        nStubsInsideEtaPhi++;
+      if (insideEta)
+        nStubsInsideEta++;
+      if (insidePhi)
+        nStubsInsidePhi++;
+    }
+  }
+
+  // Digitize a floating point number to 2s complement integer, dropping anything after the decimal point. (Kristian Harder)
+
+  Long64_t Sector::forceBitWidth(const float value, const UInt_t nBits) const {
```

Comment:
Please use standard typedefs instead of the ones from ROOT (everywhere below)
```suggestion
  int64_t Sector::forceBitWidth(const float value, const UInt_t nBits) const {
```

Reply:
done